server/server/src/billing/app_store_client.rs

```diff
@@ -57,69 +59,107 @@ pub async fn get_sub_status(
 ) -> Result<(LastTransactionItem, TransactionInfo), ServerError<UpgradeAccountAppStoreError>> {
     let resp = gen_auth_req(
         config,
-        client.get(format!("{}/{}", config.sub_statuses_url, original_transaction_id)),
+        client.get(format!("{}/{}", SUB_STATUS_PROD, original_transaction_id)),
     )?
     .send()
     .await?;
 
     let resp_status = resp.status().as_u16();
     match resp_status {
         200 => {
-            debug!("Successfully retrieved subscription status");
+            debug!("Successfully retrieved subscription status from production apple url");
 
             let sub_status: SubsStatusesResponse = resp.json().await?;
 
             for sub_group in &sub_status.data {
                 if sub_group.sub_group == config.monthly_sub_group_id {
-                    let last_trans = sub_group
-                        .last_transactions
-                        .get(0)
-                        .ok_or(ClientError(UpgradeAccountAppStoreError::InvalidAuthDetails))?;
-
-                    let part = <&str>::clone(
-                        last_trans
-                            .signed_transaction_info
-                            .split('.')
-                            .collect_vec()
-                            .get(1)
-                            .ok_or_else::<ServerError<UpgradeAccountAppStoreError>, _>(|| {
-                                internal!(
-                                    "There should be a payload in apple jwt: {:?}",
-                                    sub_status
-                                )
-                            })?,
-                    );
-
-                    let trans_info =
-                        serde_json::from_slice(&base64::decode(part).map_err::<ServerError<
-                            UpgradeAccountAppStoreError,
-                        >, _>(
-                            |err| {
-                                internal!(
-                                    "Cannot decode apple jwt payload: {:?}, err: {:?}",
-                                    part,
-                                    err
-                                )
-                            },
-                        )?)?;
-
-                    return Ok((last_trans.clone(), trans_info));
+                    return get_trans(&sub_status, sub_group);
                 }
             }
 
-            Err(internal!("No usable data returned from apple subscriptions statuses endpoint despite assumed match. resp_body: {:?}, monthly_sub_group: {}", sub_status, config.monthly_sub_group_id))
+            Err(internal!("No usable data returned from apple's production subscriptions statuses endpoint despite assumed match. resp_body: {:?}, monthly_sub_group: {}", sub_status, config.monthly_sub_group_id))
         }
         400 | 404 => {
-            let resp_body = resp.text().await.ok();
+            let error: ErrorBody = resp.json().await?;
+
+            if error.error_code == 4040005 {
```

Comment:
thought you said it was: 21007

Reply:
That was for the "verify receipt" endpoint, but no error code like that exists for the "get all subscription statuses" endpoint which we now use. This error code signifies that the original transaction id was not found, so when that happens I check sandbox.

The way this implementation now runs is that I will first check production for a subscription, then sandbox.

I tested this myself using a sandbox account, verifying it successfully makes the jump to check sandbox.